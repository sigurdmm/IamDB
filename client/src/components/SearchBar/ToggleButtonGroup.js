import React, { PureComponent } from 'react';
import './ToggleButtonGroup.less';
import PropTypes from 'prop-types';

const ToggleButton = ({ children, onClick, active }) => <button
  onClick={onClick}
  className={`togglebuttons__button${active ? ' togglebuttons__button--toggled' : ''}`}>{children}</button>;

ToggleButton.propTypes = {
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired,
};

export default class ToggleButtonGroup extends PureComponent {
  static propTypes = {
    toggled: PropTypes.number.isRequired,
    buttons: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    })),
    onToggle: PropTypes.func.isRequired,
  };

  onChange = id => () => {
    if (id === this.props.toggled) {
      return;
    }
    this.props.onToggle(id);
  };

  render() {
    return <div className="togglebuttons">
      {this.props.buttons.map((button, i) => <ToggleButton
        key={`togglebutton-${i}`}
        active={this.props.toggled === i}
        onClick={this.onChange(i)}>{button.label}</ToggleButton>)}
    </div>;
  }
}
