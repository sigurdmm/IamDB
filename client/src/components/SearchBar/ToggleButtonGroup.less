@import '../../constants';

.togglebuttons__container {
  display: block;

  position: relative;
}

.togglebuttons__container :first-child {
  border-radius: 2px 0 0 2px;
  border-right: none;
}

.togglebuttons__container :last-child {
  border-radius: 0 2px 2px 0;
}

.toggleButton {
  font-size: 1rem;
  line-height: 1.375rem;
  font-weight: 400;
  justify-content: center;
  align-items: center;
  background-color: @white;
  border: 1px solid @borderColor;
  cursor: pointer;
  min-width: 6.25rem;
  height: 2rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

button {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
}

.toggleButtonGroup__toggled {
  background-color: @primaryColor;
  color: @white;
  border-color: @primaryColor;
}
