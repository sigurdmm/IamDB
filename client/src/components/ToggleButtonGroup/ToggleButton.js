import PropTypes from 'prop-types';
import React from 'react';

const ToggleButton = ({ children, onClick, active }) => <button
  onClick={onClick}
  className={`togglebuttons__button${active ? ' togglebuttons__button--toggled' : ''}`}>{children}</button>;

ToggleButton.propTypes = {
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired,
};

export default ToggleButton;
