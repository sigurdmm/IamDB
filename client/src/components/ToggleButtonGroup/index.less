@import '../../constants';

.togglebuttons{
  display: block;
  position: relative;

  :first-child {
    border-radius: 2px 0 0 2px;
    border-right: none;
  }

  :last-child {
    border-radius: 0 2px 2px 0;
    border-right: 1px solid @borderColor;
  }

  &__button {
    &--toggled {
      background-color: @primaryColor;
      color: @white;
      border-color: @primaryColor;
    }

    justify-content: center;
    align-items: center;

    font-size: 1rem;
    line-height: 1.375rem;
    font-weight: 400;
    min-width: 6.25rem;
    height: 2rem;
    padding-left: 1rem;
    padding-right: 1rem;
    background-color: @white;
    border: 1px solid @borderColor;
    border-right: none;
    cursor: pointer;
  }
}
