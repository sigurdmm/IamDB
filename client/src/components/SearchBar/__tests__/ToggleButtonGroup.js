import React from 'react';
import { shallow } from 'enzyme';

import ToggleButtonGroup from '../ToggleButtonGroup';

const mockedState = {
  toggled: 0,
  buttons: [{ content: 'Movie' }, { content: 'TV Show' }],
  onToggle: jest.fn(),
};

describe('<ToggleButtonGroup/>', () => {
  it('should load properly', () => {
    const wrapper = shallow(<ToggleButtonGroup
      toggled={mockedState.toggled}
      onToggle={mockedState.onToggle}
      buttons={mockedState.buttons}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onToggle on change', () => {
    const spy = jest.fn();
    const wrapper = shallow(<ToggleButtonGroup
      toggled={mockedState.toggled}
      onToggle={spy}
      buttons={mockedState.buttons}/>);
    wrapper.find('ToggleButton').at(1).simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('should not call onToggle if already selected', () => {
    const spy = jest.fn();

    const wrapper = shallow(<ToggleButtonGroup
      toggled={0}
      onToggle={spy}
      buttons={mockedState.buttons}
    />);

    wrapper.find('ToggleButton').at(0).simulate('click');

    expect(spy).not.toHaveBeenCalled();
  });
});
