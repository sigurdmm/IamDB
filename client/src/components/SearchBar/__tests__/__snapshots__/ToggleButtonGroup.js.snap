// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButtonGroup/> should load properly 1`] = `
<div
  className="togglebuttons"
>
  <ToggleButton
    active={true}
    key="togglebutton-0"
    onClick={[Function]}
  >
    Movie
  </ToggleButton>
  <ToggleButton
    active={false}
    key="togglebutton-1"
    onClick={[Function]}
  >
    TV Show
  </ToggleButton>
</div>
`;
